apps/api/v2/src/ee/bookings/2024-08-13/services/bookings.service.ts

```diff
@@ -712,6 +712,47 @@ export class BookingsService_2024_08_13 {
   }
 
   async cancelBooking(request: Request, bookingUid: string, body: CancelBookingInput) {
+    try {
+      await this.canCancelBooking(bookingUid, body);
+      const bookingRequest = await this.inputService.createCancelBookingRequest(request, bookingUid, body);
+      const res = await handleCancelBooking({
+        bookingData: bookingRequest.body,
+        userId: bookingRequest.userId,
+        arePlatformEmailsEnabled: bookingRequest.arePlatformEmailsEnabled,
+        platformClientId: bookingRequest.platformClientId,
+        platformCancelUrl: bookingRequest.platformCancelUrl,
+        platformRescheduleUrl: bookingRequest.platformRescheduleUrl,
+        platformBookingUrl: bookingRequest.platformBookingUrl,
+      });
+
+      if (!res.onlyRemovedAttendee) {
+        await this.billingService.cancelUsageByBookingUid(res.bookingUid);
+      }
+
+      if ("cancelSubsequentBookings" in body && body.cancelSubsequentBookings) {
+        return this.getAllRecurringBookingsByIndividualUid(bookingUid);
+      }
+      return this.getBooking(bookingUid);
+    } catch (error) {
+      this.errorsBookingsService.handleBookingError(error, false);
```

Comment:
We are now handling all errors through _handleBookingError_ which wasn't the case earlier? This needs a thorough verification that for anyone calling _cancelBooking_ and for the users expecting a particular API response the behaviour remains unchanged.